jerry-core/include/jerryscript-port.h

```diff
@@ -138,6 +139,50 @@ double jerry_port_get_current_time (void);
  */
 struct jerry_instance_t *jerry_port_get_current_instance (void);
 
+/*
+ * Debugger Port API
+ */
+
+/**
+ * A forward declaration of various debugger structures.
+ */
+typedef struct jerry_debugger_transport_t jerry_debugger_transport_t;
+typedef struct jerry_debugger_config_t jerry_debugger_config_t;
+typedef struct jerry_debugger_conn_t jerry_debugger_conn_t;
+
+#ifdef JERRY_DEBUGGER
+
+/**
+ * Debugger connection errors.
+ */
+typedef enum {
+  JERRY_CONN_ERROR_NONE,    /**< no errors */
+  JERRY_CONN_ERROR_AGAIN,   /**< try again */
+  JERRY_CONN_ERROR_INVALID, /**< invalid argument */
+  JERRY_CONN_ERROR_IO,      /**< network I/O error */
+  JERRY_CONN_ERROR_UNKNOWN  /**< unknown error */
+} jerry_debugger_conn_errors_t;
+
+/**
+ * Transport APIs, different transport will define
+ * the real implementation of these APIs
+ */
+struct jerry_debugger_transport_t {
+  jerry_debugger_config_t *config_p;
+  jerry_debugger_conn_t* (*accept) (jerry_debugger_transport_t *transport_p);
+  jerry_debugger_conn_errors_t (*send) (jerry_debugger_conn_t *connection_p, const void *data_p, size_t data_len, ssize_t *bytes_sent);
+  jerry_debugger_conn_errors_t (*receive) (jerry_debugger_conn_t *connection_p, void *data_p, size_t max_len, ssize_t *bytes_received);
+  jerry_debugger_conn_errors_t (*close) (jerry_debugger_conn_t *connection_p);
+};
+
+/**
+ * The default transport layer that uses TCP socket
+ * other transports can be added like usb, bluetooth
+ */
+jerry_debugger_transport_t *jerry_debugger_init_socket_transport(uint16_t tcp_port);
```

Comment:
Should this be in `jerryscript-port.h`?
It's specific to the socket transport implementation, no?